examples/oauth/app/routes/login.tsx

```diff
@@ -0,0 +1,59 @@
+import type { ActionFunction, LoaderFunction } from 'remix'
+import { Form, json, useLoaderData } from 'remix'
+import { authenticator, sessionStorage, supabaseStrategy } from '~/auth.server'
+import { signInWithGithub } from '~/supabase.client'
+
+type LoaderData = {
+  error: { message: string } | null
+}
+
+export const action: ActionFunction = async({ request }) => {
+  await authenticator.authenticate('sb', request, {
+    successRedirect: '/private',
+    failureRedirect: '/login',
+  })
+}
+
+export const loader: LoaderFunction = async({ request }) => {
+  await supabaseStrategy.checkSession(request, {
+    successRedirect: '/private',
+  })
+
+  const session = await sessionStorage.getSession(
+    request.headers.get('Cookie'),
+  )
+
+  const error = session.get(
+    authenticator.sessionErrorKey,
+  ) as LoaderData['error']
+
+  return json<LoaderData>({ error })
+}
+
+export default function Screen() {
+  const { error } = useLoaderData<LoaderData>()
+
+  return (
+    <>
+      <Form method="post">
```

Comment:
can remove this form and only have the github OAuth button

Reply:
Done :)